rust/codec-lexical/src/inlines.rs

```diff
@@ -0,0 +1,202 @@
+        Link(link) => link_to_lexical(link),
+
+        _ => {
+            context.losses.add(inline.node_type().to_string());
+            lexical::InlineNode::Text(lexical::TextNode::default())
+        }
+    }
+}
+
+fn text_from_lexical(format: lexical::TextFormat, value: String) -> Inline {
+    if format.contains(TextFormat::CODE) {
+        return Inline::CodeInline(CodeInline::new(value.into()));
+    }
+
+    let mut inline = Inline::Text(Text::from(value));
+
+    if format.contains(TextFormat::BOLD) {
+        inline = Inline::Strong(Strong::new(vec![inline]))
+    }
+    if format.contains(TextFormat::ITALIC) {
+        inline = Inline::Emphasis(Emphasis::new(vec![inline]))
+    }
+    if format.contains(TextFormat::STRIKETHROUGH) {
+        inline = Inline::Strikeout(Strikeout::new(vec![inline]))
+    }
+    if format.contains(TextFormat::UNDERLINE) {
+        inline = Inline::Underline(Underline::new(vec![inline]))
+    }
+    if format.contains(TextFormat::SUBSCRIPT) {
+        inline = Inline::Subscript(Subscript::new(vec![inline]))
+    }
+    if format.contains(TextFormat::SUPERSCRIPT) {
+        inline = Inline::Superscript(Superscript::new(vec![inline]))
+    }
+    if format.contains(TextFormat::HIGHLIGHT) {
+        inline = Inline::Annotation(Annotation::new(vec![inline]))
+    }
+
+    inline
+}
+
+fn formatted_to_lexical(
+    format: lexical::TextFormat,
+    inlines: &[Inline],
+    context: &mut LexicalEncodeContext,
+) -> Vec<lexical::InlineNode> {
+    // Add the format to the context so it is applied to child inlines
+    context.text_format.insert(format);
+
+    let inlines = inlines_to_lexical(inlines, context);
+
+    // Remove the format from the context
+    context.text_format.remove(format);
+
+    inlines
+}
+
+fn text_to_lexical(text: &Text, context: &mut LexicalEncodeContext) -> lexical::InlineNode {
+    let format = context.text_format;
+    let text = text.value.to_string();
+
+    match context.format {
+        Format::Koenig => lexical::InlineNode::ExtendedText(lexical::ExtendedTextNode {
+            format,
+            text,
+            ..Default::default()
+        }),
+        _ => lexical::InlineNode::Text(lexical::TextNode {
+            format,
+            text,
+            ..Default::default()
+        }),
+    }
+}
+
+fn link_from_lexical(link: lexical::LinkNode) -> Inline {
+    if link.target.is_some() {
+        Inline::Link(Link {
+            rel: link.rel,
+            target: link.target.unwrap_or_default(),
+            content: vec![Inline::Text(Text::from(link.url))],
+            ..Default::default()
+        })
+    } else {
+        Inline::Link(Link {
+            rel: link.rel,
+            target: link.url.clone(),
+            content: vec![Inline::Text(Text::from(link.url))],
+            ..Default::default()
+        })
+    }
+}
+
+fn link_to_lexical(link: &Link) -> lexical::InlineNode {
+    if let Inline::Text(Text { value, .. }) = link.content.first().unwrap() {
```

Comment:
Avoid `.unwrap()` like the plauge - this would cause a `panic` if `link.content` is empty!

```suggestion
    if let Some(Inline::Text(Text { value, .. })) = link.content.first() {
```